starky2/src/all_stark.rs

```diff
@@ -0,0 +1,192 @@
+use std::marker::PhantomData;
+
+use plonky2::field::extension_field::{Extendable, FieldExtension};
+use plonky2::field::packed_field::PackedField;
+use plonky2::hash::hash_types::RichField;
+
+use crate::config::StarkConfig;
+use crate::constraint_consumer::{ConstraintConsumer, RecursiveConstraintConsumer};
+use crate::cross_table_lookup::CrossTableLookup;
+use crate::permutation::PermutationPair;
+use crate::stark::Stark;
+use crate::vars::{StarkEvaluationTargets, StarkEvaluationVars};
+
+#[derive(Clone)]
+pub struct AllStark<F: RichField + Extendable<D>, const D: usize> {
+    pub cpu_stark: CpuStark<F, D>,
+    pub keccak_stark: KeccakStark<F, D>,
+    pub cross_table_lookups: Vec<CrossTableLookup<F>>,
+}
+
+impl<F: RichField + Extendable<D>, const D: usize> AllStark<F, D> {
+    pub(crate) fn nums_permutation_zs(&self, config: &StarkConfig) -> Vec<usize> {
+        vec![
+            self.cpu_stark.num_permutation_batches(config),
+            self.keccak_stark.num_permutation_batches(config),
+        ]
+    }
+}
+
+#[derive(Copy, Clone)]
+pub struct CpuStark<F, const D: usize> {
+    #[allow(dead_code)]
+    num_rows: usize,
+    f: PhantomData<F>,
+}
+
+#[derive(Copy, Clone)]
+pub struct KeccakStark<F, const D: usize> {
```

Comment:
Would you mind moving these to `cpu` and `keccak` modules/directories, as per our sync? Or we can do it as a followup